modules/apps/27-interchain-accounts/module.go

```diff
@@ -37,60 +43,70 @@ func (AppModuleBasic) RegisterLegacyAminoCodec(cdc *codec.LegacyAmino) {
 	types.RegisterLegacyAminoCodec(cdc)
 }
 
+// RegisterInterfaces registers module concrete types into protobuf Any
+func (AppModuleBasic) RegisterInterfaces(registry codectypes.InterfaceRegistry) {
+	types.RegisterInterfaces(registry)
+}
+
+// DefaultGenesis returns default genesis state as raw bytes for the IBC
+// interchain accounts module
 func (AppModuleBasic) DefaultGenesis(cdc codec.JSONCodec) json.RawMessage {
 	return cdc.MustMarshalJSON(types.DefaultGenesis())
 }
 
+// ValidateGenesis performs genesis state validation for the IBC interchain acounts module
 func (AppModuleBasic) ValidateGenesis(cdc codec.JSONCodec, config client.TxEncodingConfig, bz json.RawMessage) error {
 	return nil
 }
 
+// RegisterRESTRoutes implements AppModuleBasic interface
 func (AppModuleBasic) RegisterRESTRoutes(ctx client.Context, rtr *mux.Router) {
-	// noop
 }
 
+// RegisterGRPCGatewayRoutes registers the gRPC Gateway routes for the interchain accounts module.
+func (AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *runtime.ServeMux) {
+}
+
+// GetTxCmd implements AppModuleBasic interface
 func (AppModuleBasic) GetTxCmd() *cobra.Command {
-	// noop
 	return nil
 }
 
+// GetQueryCmd implements AppModuleBasic interface
 func (AppModuleBasic) GetQueryCmd() *cobra.Command {
-	return cli.GetQueryCmd()
-}
-
-// RegisterInterfaces registers module concrete types into protobuf Any.
-func (AppModuleBasic) RegisterInterfaces(registry codectypes.InterfaceRegistry) {
-	types.RegisterInterfaces(registry)
-}
-
-// RegisterGRPCGatewayRoutes registers the gRPC Gateway routes for the interchain accounts module.
-func (AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *runtime.ServeMux) {
+	return nil
 }
 
+// AppModule is the application module for the IBC interchain accounts module
 type AppModule struct {
 	AppModuleBasic
-	keeper keeper.Keeper
+	controllerKeeper *controllerkeeper.Keeper
+	hostKeeper       *hostkeeper.Keeper
 }
 
-// NewAppModule creates an interchain accounts app module.
-func NewAppModule(k keeper.Keeper) AppModule {
+// NewAppModule creates a new IBC interchain accounts module
+func NewAppModule(controllerKeeper *controllerkeeper.Keeper, hostKeeper *hostkeeper.Keeper) AppModule {
```

Comment:
Both keepers are pointer types to allow for nil checking below in `InitGenesis` and `ExportGenesis`. 
You mentioned it may be okay to keep these as value types @colin-axner and control the enabling/disabling of controller/host via gov params, right?

Reply:
@colin-axner does that mean a chain would have to pass a governance proposal in order to disable either host/controller functionality (and not be able to set this at module init)? 

I feel like it needs to be possible to set this at init time and then have the ability to change after the fact via on-chain params/gov prop.